test/v3/createPool/gyroECLP/gyroECLP.validation.test.ts

```diff
@@ -0,0 +1,342 @@
+            expect(() =>
+                buildCallWithModifiedInput({
+                    derivedEclpParams: {
+                        tauAlpha: {
+                            x: 0n,
+                            y: _ONE_XP - _DERIVED_TAU_NORM_ACCURACY_XP - 1n,
+                        },
+                    },
+                }),
+            ).toThrowError('RotationVectorNotNormalized()');
+
+            expect(() =>
+                buildCallWithModifiedInput({
+                    derivedEclpParams: {
+                        tauAlpha: {
+                            x: 0n,
+                            y: _ONE_XP + _DERIVED_TAU_NORM_ACCURACY_XP + 1n,
+                        },
+                    },
+                }),
+            ).toThrowError('RotationVectorNotNormalized()');
+        });
+
+        test('Derived parameters u, v, w, z limits', async () => {
+            expect(() =>
+                buildCallWithModifiedInput({
+                    derivedEclpParams: { u: _ONE_XP + 1n },
+                }),
+            ).toThrowError(`u must be <= ${_ONE_XP}`);
+
+            expect(() =>
+                buildCallWithModifiedInput({
+                    derivedEclpParams: { v: _ONE_XP + 1n },
+                }),
+            ).toThrowError(`v must be <= ${_ONE_XP}`);
+
+            expect(() =>
+                buildCallWithModifiedInput({
+                    derivedEclpParams: { w: _ONE_XP + 1n },
+                }),
+            ).toThrowError(`w must be <= ${_ONE_XP}`);
+
+            expect(() =>
+                buildCallWithModifiedInput({
+                    derivedEclpParams: { z: _ONE_XP + 1n },
+                }),
+            ).toThrowError(`z must be <= ${_ONE_XP}`);
+        });
+
+        test('DerivedDsqWrong()', async () => {
+            expect(() =>
+                buildCallWithModifiedInput({
+                    derivedEclpParams: {
+                        dSq: _ONE_XP - _DERIVED_DSQ_NORM_ACCURACY_XP - 1n,
+                    },
+                }),
+            ).toThrowError('DerivedDsqWrong()');
+
+            expect(() =>
+                buildCallWithModifiedInput({
+                    derivedEclpParams: {
+                        dSq: _ONE_XP + _DERIVED_DSQ_NORM_ACCURACY_XP + 1n,
+                    },
+                }),
+            ).toThrowError('DerivedDsqWrong()');
+        });
+
+        // TODO: grow bigger brain to figure out param values that trigger InvariantDenominatorWrong
+        test.skip('InvariantDenominatorWrong()', async () => {
+            expect(
+                buildCallWithModifiedInput({
+                    eclpParams: {
+                        s: _ONE,
+                    },
+                    derivedEclpParams: {
+                        u: _ONE_XP,
+                    },
+                }),
+            ).toThrowError(
+                `InvariantDenominatorWrong: mulDenominator must be <= ${_MAX_INV_INVARIANT_DENOMINATOR_XP}`,
```

Comment:
If anyone thinks this is worth figuring out, I can spend more time on it, but kinda tricky with the derived params being calculated off chain by closed source code 

https://github.com/balancer/balancer-v3-monorepo/blob/56a0d28c39060f7b846a5b9413ec61210c03a540/pkg/interfaces/contracts/pool-gyro/IGyroECLPPool.sol#L115-L137

Reply:
Also is it weird to have the validation tests in SDK repo but the logic in balancer maths?